### DISCUSSION TITLE
v3.0.0-rc.9

### DISCUSSION THREAD

**Discussion by pi0** [nuxt/nuxt#18390] [Announcements]
**Release notes:** [v3.0.0-rc.9](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.9)

**Highlights:**

- ✅ `app.config.ts` with HMR and Reactivity support
- ✅ Updated to [nitropack@0.5.x](https://github.com/unjs/nitro/releases/tag/v0.5.0)
- ✅ Vite-Node enabled by default ([#6217](https://github.com/nuxt/framework/pull/6217))
- ⚠️ `autoImports` option and hooks is deprecated and renamed to `imports` ([#6864](https://github.com/nuxt/framework/pull/6864))  ([#7158](https://github.com/nuxt/framework/pull/7158))
- ⚠️ Vue dependency is now externalized ([#6868](https://github.com/nuxt/framework/pull/6868))
- ⚠️  Handle prerelease constraint ([#7116](https://github.com/nuxt/framework/pull/7116))
  - 👉 If you see an error like `Nuxt version ^3.0.0 is required but currently using 3.0.0-rc.9`, please contact the module author.

**Comment by AndreyYolkin** [comment]
Tried with bridge and got bug with Nuxt compatibility check. I will open issue and PR in addition to this reply
Shortly: 
https://github.com/nuxt/framework/blob/main/packages/kit/src/compatibility.ts#L14
![image](https://user-images.githubusercontent.com/11289484/188270501-5ddffb92-bce8-4a48-a5d8-2d0c7d1bc283.png)

**Reply by pi0** [reply]
Thanks for the notice! I think fixing issue with `nuxt-edge` releases instead to use same 7 digit naming for consistency and also hot fix for bridge. BTW PR welcome to fix regex accepting 7 or 8 (search in code, there are two places using `{7}`)

**Reply by AndreyYolkin** [reply]
yes, I'm currently working on it 💪

**Comment by kstraszewski** [comment]
Hi I'm getting couple of warnings like this: 
```
 WARN  Module nuxt-jsonld is disabled due to incompatibility issues:                                                                                              19:39:56
 - [nuxt] Nuxt version `^3.0.0` is required but currently using `3.0.0-rc.9`
```

Is there any way to enable them?

**Reply by pi0** [reply]
PR already made by @danielroe  https://github.com/ymmooot/nuxt-jsonld/pull/952

**Reply by kstraszewski** [reply]
Love you guys!!

**Comment by kstraszewski** [comment]
After building and serving an app I am getting this errors now and CSS does not load at all. The CSS files are building properly.
<img width="436" alt="image" src="https://user-images.githubusercontent.com/10195995/188283757-0ad57ffa-ff9b-4617-b0af-bf9a186e696b.png">

**Reply by danielroe** [reply]
Would you open a new issue with a reproduction? 🙏

**Reply by kstraszewski** [reply]
ofc, but tomorrow :) But it is a regular production build and serve. It might be vuetify fault coz it loads lots of css

**Reply by kstraszewski** [reply]
Hi @danielroe I've posted the issue https://github.com/nuxt/framework/issues/7217

**Comment by some-user123** [comment]
`useFetch<T>` is now retuning `data` attribute with type `Ref<T | null>` instead of `Ref<T>`. Similarly for `useAsyncData`. Is that intended? Breaks code that relies on the type...

In principle that makes sense, because `data` will be `null` if an error occured. But narrowing down the type seems to only apply in `<script>` not to the template (cf. https://stackblitz.com/edit/nuxt-starter-efngmc?file=app.vue)

**Comment by SnowGuest** [comment]
![image](https://user-images.githubusercontent.com/59474029/188366296-44fb7c4b-2839-4619-978b-98b589750225.png)
![image](https://user-images.githubusercontent.com/59474029/188366383-7f761496-e31b-4301-ae5c-640c489c8fad.png)

I am confused by this error.
He didn't tell me at all what module was missing.
Project is normal in Nuxt Rc.8.

**Reply by pi0** [reply]
We are working to improve vite-node error handling (it is enabled by default in rc.9). Can you please open an issue with minimum step to reproduce? Should be easily solvable :)

**Reply by SnowGuest** [reply]
I have tried to reproduce this problem, but I did not have this error when I created the RC.9 template. I will try my best to find the smallest reproduction template

**Reply by SnowGuest** [reply]
![image](https://user-images.githubusercontent.com/59474029/188392109-51e87a29-f9b7-47ae-8b53-57d8ce4be452.png)

I also have a problem with peer dependencies.
And after I regenerated the node_modules folder, there was also an error that defu dependencies were not found.

**Reply by pi0** [reply]
What is your package manager?

**Reply by SnowGuest** [reply]
I am using pnpm

**Reply by pi0** [reply]
For pnpm, you should use `pnpm install --shamefully-hoist`

**Reply by SnowGuest** [reply]
After I switched to the command just now, the previous problems all disappeared.
thanks for your answer.

**Reply by benoitdemaegdt** [reply]
Hey there 👋 

I have the [exact same error](https://github.com/benoitdemaegdt/monpetitsommet/pull/279), but using `npm`.
<img width="288" alt="Capture d’écran 2022-09-05 à 19 29 52" src="https://user-images.githubusercontent.com/25749578/188495540-5dec4c20-1428-46da-a044-d5ef4d19fdf1.png">

Any idea about how to get more details (which module is not found ?) ?

**Reply by eruhaji** [reply]
I have the same issue, I'm using npm though so I can't shamefully hoist ;)

**Reply by Mabi19** [reply]
I have the same issue, but with Yarn (1.22.17). My dependencies (other than Nuxt) are some packages for DB access and utilities on the server-side and Tiptap with the Vue 3 integration.

**Reply by benoitdemaegdt** [reply]
I only have this issue when I run `npm run dev`
=> `npm run generate` and `npm run build` are ok, no error 🤔

**Reply by raducretu** [reply]
i have the same error "ERR_MODULE_NOT_FOUND" on npm, and it's very strange it's only happens on my personal laptop, on my office iMac it works

**Reply by benoitdemaegdt** [reply]
That is very strange indeed. Here is what I tried so far :
1/ Removing few nuxt modules and see if i get rid of this error => no success 🔴 
2/ Start a new nuxt project and add some of these modules => no error, it works.

I'll try to clone and run the project on another computer 🤔

**Reply by raducretu** [reply]
i tried edge channel also, using different node version, removing all modules, same thing:
- Operating System: `Darwin`
- Node Version:     `v16.13.1`
- Nuxt Version:     `3.0.0-rc.10-27718444.40d0907`
- Nitro Version:    `0.5.2-27718085.c24dbcf`
- Package Manager:  `npm@8.3.0`
- Builder:          `vite`
- User Config:      `publicRuntimeConfig`, `buildModules`, `css`, `build`, `modules`, `intlify`
- Runtime Modules:  `@intlify/nuxt3@0.2.4`, `@vueuse/nuxt@9.2.0`
- Build Modules:    `@pinia/nuxt@0.4.2`, `@nuxtjs/tailwindcss@5.3.3`

**Comment by sweetroll** [comment]
![Screen Shot 2022-09-05 at 10 14 31 am](https://user-images.githubusercontent.com/462429/188401529-8f646445-e0e2-48a8-ae13-39df3cb123e3.png)

Fresh install using npx nuxi init on Node 16.17.0

**Reply by engenes** [reply]
same problem
reproduction [here](https://github.com/nuxt/framework/discussions/7239)

**Reply by bkuri** [reply]
I have a similar error, but it also seems unable to resolve local https certificates (needed for storyblok development):

![2022-09-05T20:09:20,876945548-05:00](https://user-images.githubusercontent.com/138886/188526429-acf4eea0-d200-40bb-8fbf-719075f5225c.png)

In my case I upgraded from `rc.8` and am using node `v16.16.0`.

I also just tried it under `v18.8.0` and got a (slightly) different error:
![2022-09-05T20:26:05,792451830-05:00](https://user-images.githubusercontent.com/138886/188527769-5ae41f77-45f3-495d-848e-4c3c772735bd.png)

**Reply by Lucky-Strike1695** [reply]
Same here, 500 error on localhost https (upgraded from rc8).

**Reply by cwe-dev** [reply]
same problems. it's but from https dev
related to https://github.com/unjs/ohmyfetch/issues/120
init bug -https://github.com/nuxt/framework/issues/7468

**Reply by danielroe** [reply]
FYI, this was resolved in the edge channel a week ago via https://github.com/nuxt/framework/pull/7271.

**Reply by cwe-dev** [reply]
@danielroe i tried use edge, but problem not a solved( ill wait rc10 :D

**Reply by danielroe** [reply]
if it's not solved then please open an issue so we can fix it for rc10 😉

**Comment by konradsienkowski** [comment]
Hey, after updating to the RC 9 running `npm run dev` works fine, but trying to use `npm run build` and preview the build results with an error in the console:
`Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.`

None of the nuxt.config.ts proposed solutions mentioned in [that issue](https://github.com/nuxt/framework/issues/5303) works in my case. I was using node v16.15.1, updating to v16.17.0 didn't help

**Reply by danielroe** [reply]
Would you raise a new issue with a minimal reproduction? It sounds like a possible mismatch between your `nuxt` + `@nuxt/vite-builder` versions.

**Reply by konradsienkowski** [reply]
I'm trying to, but it's quite a complex app so trimming it down takes some time. Based on node_modules content, the `@nuxt/vite-builder` version specified in its `package.json` is also `3.0.0-rc.9`.

**Reply by konradsienkowski** [reply]
@danielroe issue is here: https://github.com/nuxt/framework/issues/7285

**Comment by Silbad** [comment]
Hello,
I have a new warning with rc9 : 
` WARN  autoImports:extend hook is deprecated. Use addImports() from @nuxt/kit or imports:extend with nuxt>=3.0.0-rc.9. `
any idea ?
Thanks

**Reply by danielroe** [reply]
This likely comes from a module. You can ignore for now, but if you can tell which module it is, feel free to open an issue to tell them.

**Comment by raducretu** [comment]
I updated a RC8 to RC9 and I get this error:

`
 ERROR  Error while formatting vite error: {
  pos: 2431,
  loc: {
    line: 64,
    column: 4
  },
  raisedAt: 2432,
  plugin: 'nuxt:composable-keys',
  id: '/Users/raducretu/Work/keybox-admin/node_modules/nuxt/dist/pages/runtime/router.mjs',
  pluginCode: 'import { computed, reactive, shallowRef } from "vue";\n' +
    'import {\n' +
    '  createRouter,\n' +
    '  createWebHistory,\n' +
    '  createMemoryHistory\n' +
    '} from "vue-router";\n' +
    'import { createError } from "h3";\n' +
    'import { withoutBase, isEqual } from "ufo";\n' +
    'import NuxtPage from "./page.mjs";\n' +
    'import { callWithNuxt, defineNuxtPlugin, useRuntimeConfig, showError, clearError, navigateTo, useError, useState } from "#app";\n' +
    'import routes from "#build/routes";\n' +
    'import routerOptions from "#build/router.options";\n' +
    'import { globalMiddleware, namedMiddleware } from "#build/middleware";\n' +
    'function createCurrentLocation(base, location) {\n' +
    '  const { pathname, search, hash } = location;\n' +
    '  const hashPos = base.indexOf("#");\n' +
    '  if (hashPos > -1) {\n' +
    '    const slicePos = hash.includes(base.slice(hashPos)) ? base.slice(hashPos).length : 1;\n' +
    '    let pathFromHash = hash.slice(slicePos);\n' +
    '    if (pathFromHash[0] !== "/") {\n' +
    '      pathFromHash = "/" + pathFromHash;\n' +
    '    }\n' +
    '    return withoutBase(pathFromHash, "");\n' +
    '  }\n' +
    '  const path = withoutBase(pathname, base);\n' +
    '  return path + search + hash;\n' +
    '}\n' +
    'export default defineNuxtPlugin(async (nuxtApp) => {\n' +
    '  nuxtApp.vueApp.component("NuxtPage", NuxtPage);\n' +
    '  nuxtApp.vueApp.component("NuxtNestedPage", NuxtPage);\n' +
    '  nuxtApp.vueApp.component("NuxtChild", NuxtPage);\n' +
    '  const baseURL = useRuntimeConfig().app.baseURL;\n' +
    '  const routerHistory = false ? createWebHistory(baseURL) : createMemoryHistory(baseURL);\n' +
    '  const initialURL = true ? nuxtApp.ssrContext.url : createCurrentLocation(baseURL, window.location);\n' +
    '  const router = createRouter({\n' +
    '    ...routerOptions,\n' +
    '    history: routerHistory,\n' +
    '    routes\n' +
    '  });\n' +
    '  nuxtApp.vueApp.use(router);\n' +
    '  const previousRoute = shallowRef(router.currentRoute.value);\n' +
    '  router.afterEach((_to, from) => {\n' +
    '    previousRoute.value = from;\n' +
    '  });\n' +
    '  Object.defineProperty(nuxtApp.vueApp.config.globalProperties, "previousRoute", {\n' +
    '    get: () => previousRoute.value\n' +
    '  });\n' +
    '  const _route = shallowRef(router.resolve(initialURL));\n' +
    '  const syncCurrentRoute = () => {\n' +
    '    _route.value = router.currentRoute.value;\n' +
    '  };\n' +
    '  nuxtApp.hook("page:finish", syncCurrentRoute);\n' +
    '  router.afterEach((to, from) => {\n' +
    '    if (to.matched[0]?.components?.default === from.matched[0]?.components?.default) {\n' +
    '      syncCurrentRoute();\n' +
    '    }\n' +
    '  });\n' +
    '  const route = {};\n' +
    '  for (const key in _route.value) {\n' +
    '    route[key] = computed(() => _route.value[key]);\n' +
    '  }\n' +
    '  nuxtApp._route = reactive(route);\n' +
    '  nuxtApp._middleware = nuxtApp._middleware || {\n' +
    '    {}: [],\n' +
    '    named: {}\n' +
    '  };\n' +
    '  const error = useError();\n' +
    '  try {\n' +
    '    if (true) {\n' +
    '      await router.push(initialURL);\n' +
    '    }\n' +
    '    await router.isReady();\n' +
    '  } catch (error2) {\n' +
    '    callWithNuxt(nuxtApp, showError, [error2]);\n' +
    '  }\n' +
    '  const initialLayout = useState("_layout");\n' +
    '  router.beforeEach(async (to, from) => {\n' +
    '    to.meta = reactive(to.meta);\n' +
    '    if (nuxtApp.isHydrating) {\n' +
    '      to.meta.layout = initialLayout.value ?? to.meta.layout;\n' +
    '    }\n' +
    '    nuxtApp._processingMiddleware = true;\n' +
    '    const middlewareEntries = /* @__PURE__ */ new Set([...globalMiddleware, ...nuxtApp._middleware.global]);\n' +
    '    for (const component of to.matched) {\n' +
    '      const componentMiddleware = component.meta.middleware;\n' +
    '      if (!componentMiddleware) {\n' +
    '        continue;\n' +
    '      }\n' +
    '      if (Array.isArray(componentMiddleware)) {\n' +
    '        for (const entry of componentMiddleware) {\n' +
    '          middlewareEntries.add(entry);\n' +
    '        }\n' +
    '      } else {\n' +
    '        middlewareEntries.add(componentMiddleware);\n' +
    '      }\n' +
    '    }\n' +
    '    for (const entry of middlewareEntries) {\n' +
    '      const middleware = typeof entry === "string" ? nuxtApp._middleware.named[entry] || await namedMiddleware[entry]?.().then((r) => r.default || r) : entry;\n' +
    '      if (!middleware) {\n' +
    '        if (true) {\n' +
    '          throw new Error(`Unknown route middleware: \'${entry}\'. Valid middleware: ${Object.keys(namedMiddleware).map((mw) => `\'${mw}\'`).join(", ")}.`);\n' +
    '        }\n' +
    "        throw new Error(`Unknown route middleware: '${entry}'.`);\n" +
    '      }\n' +
    '      const result = await callWithNuxt(nuxtApp, middleware, [to, from]);\n' +
    '      if (true || !nuxtApp.payload.serverRendered && nuxtApp.isHydrating) {\n' +
    '        if (result === false || result instanceof Error) {\n' +
    '          const error2 = result || createError({\n' +
    '            statusMessage: `Route navigation aborted: ${initialURL}`\n' +
    '          });\n' +
    '          return callWithNuxt(nuxtApp, showError, [error2]);\n' +
    '        }\n' +
    '      }\n' +
    '      if (result || result === false) {\n' +
    '        return result;\n' +
    '      }\n' +
    '    }\n' +
    '  });\n' +
    '  router.afterEach(async (to) => {\n' +
    '    delete nuxtApp._processingMiddleware;\n' +
    '    if (false && !nuxtApp.isHydrating && error.value) {\n' +
    '      await callWithNuxt(nuxtApp, clearError);\n' +
    '    }\n' +
    '    if (to.matched.length === 0) {\n' +
    '      callWithNuxt(nuxtApp, showError, [createError({\n' +
    '        statusCode: 404,\n' +
    '        fatal: false,\n' +
    '        statusMessage: `Page not found: ${to.fullPath}`\n' +
    '      })]);\n' +
    '    } else if (true && to.matched[0].name === "404" && nuxtApp.ssrContext) {\n' +
    '      nuxtApp.ssrContext.event.res.statusCode = 404;\n' +
    '    } else if (true) {\n' +
    '      const currentURL = to.fullPath || "/";\n' +
    '      if (!isEqual(currentURL, initialURL)) {\n' +
    '        await callWithNuxt(nuxtApp, navigateTo, [currentURL]);\n' +
    '      }\n' +
    '    }\n' +
    '  });\n' +
    '  nuxtApp.hooks.hookOnce("app:created", async () => {\n' +
    '    try {\n' +
    '      await router.replace({\n' +
    '        ...router.resolve(initialURL),\n' +
    '        name: void 0,\n' +
    '        force: true\n' +
    '      });\n' +
    '    } catch (error2) {\n' +
    '      callWithNuxt(nuxtApp, showError, [error2]);\n' +
    '    }\n' +
    '  });\n' +
    '  return { provide: { router } };\n' +
    '});\n',
  frame: '62 |    nuxtApp._route = reactive(route);\n' +
    '63 |    nuxtApp._middleware = nuxtApp._middleware || {\n' +
    '64 |      {}: [],\n' +
    '   |      ^\n' +
    '65 |      named: {}\n' +
    '66 |    };'
}



`
